app/screens/scheduled_post_options/scheduled_post_picker.tsx

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo, useState} from 'react';
+import {Platform, View} from 'react-native';
+
+import FormattedText from '@components/formatted_text';
+import {Screens} from '@constants';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import BottomSheet from '@screens/bottom_sheet';
+import ScheduledPostCoreOptions from '@screens/scheduled_post_options/core_options';
+import ScheduledPostFooter from '@screens/scheduled_post_options/footer';
+import {logInfo} from '@utils/log';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+import {getTimezone} from '@utils/user';
+
+import type {BottomSheetFooterProps} from '@gorhom/bottom-sheet';
+import type UserModel from '@typings/database/models/servers/user';
+
+const OPTIONS_PADDING = 12;
+const OPTIONS_SEPARATOR_HEIGHT = 1;
+const TITLE_HEIGHT = 54;
+const ITEM_HEIGHT = 48;
+
+export const SCHEDULED_POST_OPTIONS_BUTTON = 'close-scheduled-post-options';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        backgroundColor: theme.centerChannelBg,
+        height: 200,
+    },
+    titleContainer: {
+        alignItems: 'center',
+        flexDirection: 'row',
+    },
+    title: {
+        color: theme.centerChannelColor,
+        ...typography('Heading', 600, 'SemiBold'),
+    },
+    optionsContainer: {
+        paddingTop: OPTIONS_PADDING,
+    },
+    optionsSeparator: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        height: OPTIONS_SEPARATOR_HEIGHT,
+    },
+}));
+
+type Props = {
+    currentUser?: UserModel;
+    serverUrl: string;
+}
+
+export function ScheduledPostOptions({currentUser}: Props) {
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+
+    const [selectedTime, setSelectedTime] = useState<string | null>(null);
+
+    const userTimezone = getTimezone(currentUser?.timezone);
```

Comment:
If you only use the timezone, you can get the timezone directly from the `withDatabase`. You would have to change it to look something like this:

```
currentUserTimezone: observeCurrentUser(database).pipe(switchMap((u) => of$(u?.timezone)),
```
(Not sure if I mess some detail, but you can look at other index files for those :P )

This way, this component only re-renders when relevant changes in the user happen, and not always.

Reply:
Done